tests/test_job.py

```diff
@@ -1219,3 +1220,19 @@ def test_execution_order_with_dual_dependency(self):
         self.assertEqual(queue.count, 0)
         self.assertTrue(all(job.is_finished for job in [job_slow_1, job_slow_2, job_A, job_B]))
         self.assertEqual(jobs_completed, ["slow_1:w1", "B:w1", "slow_2:w2", "A"])
+
+    @unittest.skipIf(get_version(Redis()) < (5, 0, 0), 'Skip if Redis server < 5.0')
+    def test_blocking_result_fetch(self):
+        job_sleep_seconds = 2
+        block_seconds = 5
+        q = Queue()
+        job = q.enqueue(fixtures.long_running_job, job_sleep_seconds)
+        w = Worker([q])
+        started_at = time.time()
+        w.work(burst=True)
+        result = job.latest_result(timeout=block_seconds)
```

Comment:
This doesn't test the blocking feature because `job.latest_result` is called after job has finished executin (line 1232). To test if this truly blocks, the work has to be done in the background.

It should look like this:
```python
w.work(burst=True)  # In the background
started_at = time.time()
result = job.latest_result(timeout=block_seconds)
self.assertGreaterEqual(blocked_for, job_sleep_seconds)
```

Reply:
Thank you, good catch!

I replaced `w.work` with `fixtures.start_worker_process` which runs in the background.

(I tried moving `started_at =...` after the worker start, but it creates a flaky race-condition if the job starts before `time.time()` is called).